lib/grpperm.gi

```diff
@@ -1232,9 +1232,72 @@ end );
 ##
 #M  IsSolvableGroup( <G> )  . . . . . . . . . . . . . . . .  solvability test
 ##
+
+# fast test for finding iteratively nontrivial commutators in a permutation group.
+# if this goes on too long the group may not be solvable. This can be much faster
+# for large degree groups than to use the full pcgs machinery (which uses the same
+# idea but builds a pcgs on the way).
+BindGlobal("QuickUnsolvabilityTestPerm",function(G)
+local som,elvth,fct,gens,new,l,i,j,a,b,bound;
+  # a few moved points
+  som:=MovedPoints(G);
+  bound:=Int(LogInt(Length(som)^5,3)/2);
+  if Length(som)>100 then
+    som:=som{List([1..100],x->Random([1..Length(som)]))};
+  fi;
+
+  elvth:=One(G);
+  fct:=function(G) 
```

Comment:
`fct` seems to be a way to quickly generate pseudo random elements of `G`? If so, why can't `PseudoRandom(G)` be used?

Reply:
In some examples I tried PseudoRandom took too long, as it aims to mix better. Thus, using PseudoRandom, the supposedly quick first test took longer than the proper test using the Sims algorithm.
This (pathetic, but quick) set of products was good enough to show in the experiment cases that the group is not solvable, while in some of these cases the Sims algorithm took very long, while building a partial stabilizer chain.

Reply:
OK, makes sense, thank you for the explanation.